### PR TITLE
Add microplanning data seeding command

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description

Dev tooling only — no user-facing change.

## Technical Summary

Groundwork for the [map auto-zoom work](https://dimagi.atlassian.net/browse/CCCT-2749). `generate_sample_data` builds opportunities and visits but no work areas, so populating the microplanning maps locally meant hand-rolling CSV uploads through Celery, or an opportunity created through the real `OpportunityInit` flow with a live CommCare HQ connection.

`--opp-id` seeds work areas onto an opportunity that already exists (e.g. one `generate_sample_data` produced); with no arguments it builds a self-contained demo opportunity.

Two decisions worth knowing about:

- Clusters are placed several degrees apart. The auto-zoom behaviour being built is indistinguishable from "zoom to everything" when every work area sits in the same small box — which is what the existing `WorkAreaFactory` produces (every boundary is the same 77–78/28–29 square).

It lives in the microplanning app rather than next to `generate_sample_data` in `reports/`, since that's where the domain lives.

### Safety story

A management command with no runtime callers, so nothing in the app can reach it. It refuses to run unless `DEBUG` is on, because it creates a superuser with a well-known password; `--force` overrides.

On existing data: `--opp-id` against a real opportunity replaces that opportunity's work areas by default (`--keep-existing` to add instead). Deletion is scoped to rows hanging off a work area — visits with no work area are left alone — and the whole run is one transaction. It only sets `program_manager` on a workspace it created itself, and reports rather than mutates when pointed at one that exists.

### Automated test coverage

None, matching the two existing data-generation commands. Verified by running it repeatedly against a dev DB: idempotent across runs, `--clusters 9 --grid 2`, and a bad `--opp-id` producing a clean `CommandError`.

### QA Plan

N/A — local dev tooling, not deployed behaviour.

### Labels & Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+        if options["opp_id"]:
+            try:
+                opportunity = get_object_by_uuid_or_int(
+                    Opportunity.objects.all(), options["opp_id"], uuid_field="opportunity_id"
+                )
+            except Http404:
+                raise CommandError(f"No opportunity matches --opp-id {options['opp_id']!r}")
+            self.stdout.write(f"Using existing opportunity: {opportunity.name}")
+            return opportunity, False
+        return self.ensure_demo_opportunity(options["org_slug"]), not options["org_slug"]
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py]: Fixed in 03298f2a. Applied on merit rather than as a lint fix — this repo's ruff config selects only `["E", "F", "W", "I"]` (pyproject.toml:68), so bugbear is not enabled here and the `linter` CI job passes without it. `from None` is still the right call: it keeps the `Http404` traceback out of an error the user is meant to read.

*— Claude Code*

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+                    target_population=random.randint(200, 900),
+                    status=statuses[cell_index % len(statuses)],
+                    case_id=str(uuid.uuid4()),
+                    case_properties={"lga": f"{cluster.name} LGA"},
+                )
+            )
+
+        # bulk_create returns the rows with their pks set, and each one already holds the
+        # OpportunityAccess object create_visits() needs, so no re-read is required.
+        created = WorkArea.objects.bulk_create(pending)
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py:567]: Fixed in 86e890f6, though not as suggested — the crash is real but the diagnosis points at the wrong constraint.

`build_cluster()` creates the implementation area before the groups, and the groups before the work areas, so on a repeat `--keep-existing` run `unique_implementation_area_name_per_opportunity` raises first. `unique_name_per_opportunity` on the group is next. The work area slug is never reached, so `ignore_conflicts=True` on that `bulk_create` would sit on unreachable code — and silently swallowing slug conflicts would leave a partial seed rather than fixing anything.

The names are now generated in one place and checked before any writes, so the command fails with what to do about it instead of an IntegrityError:

```
CommandError: --keep-existing would recreate names this opportunity already has
(Central Group 1, Central Group 2, Central Implementation Area, …). Drop
--keep-existing to replace the seeded areas, or use a different --clusters count.
```

*— Claude Code*

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+                ],
+            ).values_list("name", flat=True)
+        )
+        if taken:
+            shown = ", ".join(sorted(taken)[:3])
+            more = ", …" if len(taken) > 3 else ""
+            raise CommandError(
+                f"--keep-existing would recreate names this opportunity already has ({shown}{more}). "
+                "Drop --keep-existing to replace the seeded areas, or use a different --clusters count."
+            )
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py]: Fixed in d732cf93. Real gap — and precisely the case `--keep-existing` exists for: an opportunity whose areas came from a CSV import can carry `north-wa-01` without either the implementation area name or the group name matching, so the preflight waved it through and `bulk_create` raised.

The check now covers all three per-opportunity constraints. Verified against a scratch opportunity holding only a matching slug (no implementation area, no group):

```
CommandError: --keep-existing would recreate rows this opportunity already has
(north-wa-01). Drop --keep-existing to replace the seeded areas, or use a
different --clusters count.
```

*— Claude Code*

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+            ).values_list("name", flat=True)
+        ) + list(
+            WorkAreaGroup.objects.filter(
+                opportunity=opportunity,
+                name__in=[
+                    group_name(cluster, index)
+                    for cluster in clusters
+                    for index in range(options["groups_per_cluster"])
+                ],
+            ).values_list("name", flat=True)
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py]: Fixed in d732cf93 — same gap @sentry flagged, and the reasoning here is the accurate one: an existing `north-wa-01` with neither name matching did reach `bulk_create()`.

One deviation from the proposed patch: it inlines the slug format, which is the drift risk the previous commit removed for the other two names by extracting `implementation_area_name()` and `group_name()`. Added `work_area_slug()` alongside them and used it in both the preflight and `create_cluster_areas()`, so the check and the creation site cannot disagree.

*— Claude Code*

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+                [work_area_slug(cluster, index) for cluster in clusters for index in range(options["grid"] ** 2)],
+            ),
+        ]
+        if taken:
+            shown = ", ".join(sorted(taken)[:3])
+            more = ", …" if len(taken) > 3 else ""
+            raise CommandError(
+                f"--keep-existing would recreate rows this opportunity already has ({shown}{more}). "
+                "Drop --keep-existing to replace the seeded areas, or use a different --clusters count."
+            )
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py:391]: Fixed in d7ebb69. You're right on both counts — `clear_areas()` deletes every work area on the opportunity, not just the seeded ones, and cluster 0 is in every valid `--clusters` count so raising it can never dodge a `North`/`north-wa-01` collision. The message now states the deletion scope and points at seeding a different opportunity instead.

*— Claude Code*

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 6]
```diff
@@ -0,0 +1,641 @@
...
+                },
+            )
+            deliver_unit, _ = DeliverUnit.objects.get_or_create(
+                app=opportunity.deliver_app,
+                slug=slug,
+                defaults={"payment_unit": payment_unit, "name": f"{name} DU"},
+            )
+            if deliver_unit.payment_unit_id is None:
+                deliver_unit.payment_unit = payment_unit
+                deliver_unit.save(update_fields=["payment_unit"])
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py]: Fixed in 15055f5, though not via the stated trigger. `PaymentUnit.get_or_create` is keyed on `(opportunity, name)`, so a re-run with `--keep-existing` reuses the same payment unit and the pairing stays intact — that path was never broken.

The real hole is an `--opp-id` opportunity whose deliver app already has a deliver unit on one of the fixed coverage slugs (`SERVICE_DELIVERY_UNIT_SLUG` / `NO_CHILDREN_WORK_AREA_UNIT_SLUG`). A payment unit named "Service Delivery" then gets created fresh, `payment_unit_id is None` is false, and the seeded `CompletedWork.payment_unit` ends up not owning the `UserVisit.deliver_unit` recorded against it.

Rather than unconditionally repointing, `ensure_payment_units` now resolves the deliver unit first and adopts its existing payment unit when that unit belongs to this opportunity — no mutation of real config. It only repoints when the payment unit belongs to a *different* opportunity (shared deliver app), where cross-opportunity pairing would be worse. Covered by four tests in `microplanning/tests/test_generate_microplanning_data.py`; the two edge-case ones fail on the previous code.

*— Claude Code*

File: commcare_connect/microplanning/management/commands/generate_microplanning_data.py [hunk 7]
```diff
@@ -0,0 +1,667 @@
...
+                        completed_work=completed_work,
+                        work_area=work_area,
+                        entity_id=completed_work.entity_id,
+                        entity_name=completed_work.entity_name,
+                        status=VisitValidationStatus.approved,
+                        visit_date=now - timedelta(days=random.randint(0, 90), hours=n),
+                        # The visit tile view parses this as "<lat> <lon> <alt> <accuracy>".
+                        location=f"{lat} {lon} 0 5",
+                        form_json={"seeded": True},
+                        xform_id=str(uuid.uuid4()),
```
[the remark this answers is not in the document]
Comment (snopoke) [reply commcare_connect/microplanning/management/commands/generate_microplanning_data.py:612]: Fixed in 5f443fa. Confirmed: `CompletedWork.approved_count` (models.py:872) filters on `review_status=agree`, and the real auto-approval path sets both fields together (`form_receiver/processor.py:536`), so seeded visits left at the `pending` default were unlike any genuinely approved visit and made `payment_accrued` zero.

Seeded visits now carry `review_status=VisitReviewStatus.agree`. Covered by `test_seeded_visits_accrue_payment`, which runs the command and asserts `approved_count > 0` and `payment_accrued > 0` — it fails without the change.

*— Claude Code*

**Review by pxwxnvermx — APPROVED** [review]
LGTM
